src/gui/virtualstudio.cpp

```diff
@@ -1063,6 +1062,18 @@ void VirtualStudio::handleDeeplinkRequest(const QUrl& link)
         return;
     }
 
+    // special case if on create_studio screen
+    if (m_windowState == "create_studio") {
+        refreshStudios(0, true);
+        if (showDeviceSetup()) {
+            setWindowState("setup");
+            m_audioConfigPtr->startAudio();
+        } else {
+            setWindowState("connected");
```

Comment:
I think you may need a `joinStudio()` after setting window state here.

Reply:
I found that calling `joinStudio()` directly here didn't work because the newly-created server isn't yet in `m_servers`, and `m_servers` is asynchronously updated in a network reply.

But that function is called through:
```
refreshStudios(0, true)
    -> getServerList(true, 0)
        -> emit refreshFinished(index);
            -> connect(this, &VirtualStudio::refreshFinished, this, &VirtualStudio::joinStudio,
            Qt::QueuedConnection);
```

Reply:
I believe this code is for handling deep link requests while you are on the create studio page. It's kind of an odd corner-case, but why would the deep-linked studio not be created yet? Why would we need to refresh studios at all?

Reply:
The deep-linked server does exist but the app does not know that yet. At the time you click "Create", a new server is created and the deep-link URL is fired. But the [code behind joinStudio()](https://github.com/jacktrip/jacktrip/blob/main/src/gui/virtualstudio.cpp#L563) relies on the internally-stored array of servers which is fetched by a timer or by calling refreshStudios().